### PR TITLE
[TK-05725] Kitsune Peer Gossip

### PR DESCRIPTION
**Pull request by neonphog**

### CODE REVIEW

File: crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/gossip.rs [hunk 1]
```diff
@@ -19,7 +20,7 @@ ghost_actor::ghost_chan! {
             dht_arc: DhtArc,
             since_utc_epoch_s: i64,
             until_utc_epoch_s: i64,
-        ) -> Vec<Arc<KitsuneOpHash>>;
+        ) -> (Vec<Arc<KitsuneOpHash>>, Vec<AgentInfoSigned>);
```
Comment (neonphog) [inline crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/gossip.rs:23]: Hmmm - I'm not sure we want to be sending the full signed blobs in the discovery phase... many of them both agents probably have already, so it's a lot of network load.

We might need to do something like `Vec<(Arc<KitsuneAgent>, u64)>` where u64 is the signed at time - alternately, we could do a blake2b hash of the agent_info (the portion that is signed) - but might need to wait on meister's PR that makes that a Vec<u8> for consistency.

**Comment by freesig** [comment]
I think everything is here that is needed. I couldn't get the Mem test to work so I ignored it for now assuming we can get it working in the future or if @neonphog knows how to get it working

File: crates/holochain/src/conductor/conductor.rs [hunk 2]
```diff
@@ -998,10 +1018,9 @@ mod builder {
                 tmpdir,
             } = test_env;
             let keystore = environment.keystore();
-            let (holochain_p2p, p2p_evt) = holochain_p2p::spawn_holochain_p2p(
-                holochain_p2p::kitsune_p2p::KitsuneP2pConfig::default(),
-            )
-            .await?;
+            let (holochain_p2p, p2p_evt) =
+                holochain_p2p::spawn_holochain_p2p(self.config.network.clone().unwrap_or_default())
+                    .await?;
```
Comment (neonphog) [inline crates/holochain/src/conductor/conductor.rs:1023]: hah, woops

**Review by neonphog — COMMENTED** [review]
Woot, looks awesome - I can't approve though, since I opened this PR : )

**Review by freesig — APPROVED** [review]

**Pull request merged**